lib/active_record/associated_object.rb

```diff
@@ -1,4 +1,13 @@
 class ActiveRecord::AssociatedObject
+  class Polymorphic < self
+    def self.inherited(klass)
+      if name = klass.module_parent_name
```

Comment:
I'm using/abusing the `module_parent_name` to detect `Post::Pricing` versus just `Pricing` and then only firing the connecting behavior when we're doing `Post::Pricing`. So technically we don't need a separate `Polymorphic` object to capture this and we could stash this condition in `ActiveRecord::AssociatedObject`.

Yeah, having written it out now, I actually think that's more intuitive and it simplifies our object hierarchy.